exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/join/TestHashJoinJPPDCorrectness.java

```diff
@@ -0,0 +1,184 @@
+
+import org.apache.drill.categories.SqlTest;
+import org.apache.drill.exec.ExecConstants;
+import org.apache.drill.test.ClusterFixture;
+import org.apache.drill.test.ClusterTest;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.junit.Assert.assertThat;
+import java.nio.file.Paths;
+
+@Category(SqlTest.class)
+public class TestHashJoinJPPDCorrectness extends ClusterTest {
+
+  private static final String runtimeFilterOption = "ALTER SESSION SET `" +
+    ExecConstants.HASHJOIN_ENABLE_RUNTIME_FILTER_KEY + "` = %s";
+
+  private static final String runtimeFilterWaitOption = "ALTER SESSION SET `" +
+    ExecConstants.HASHJOIN_RUNTIME_FILTER_WAITING_ENABLE_KEY + "` = %s";
+
+  private static final String runtimeFilterWaitTimeOption = "ALTER SESSION SET `" +
+    ExecConstants.HASHJOIN_RUNTIME_FILTER_MAX_WAITING_TIME_KEY + "` = %d";
+
+  @BeforeClass
+  public static void setUp() throws Exception {
+    dirTestWatcher.copyResourceToRoot(Paths.get("tpchmulti"));
+    // Reduce the slice target so that there are multiple minor fragments with exchange, otherwise RuntimeFilter
+    // will not be inserted in the plan
+    startCluster(ClusterFixture.builder(dirTestWatcher)
+      .clusterSize(2)
+      .maxParallelization(1)
+      .systemOption(ExecConstants.SLICE_TARGET, 10));
+  }
+
+  @After
+  public void tearDown() {
+    client.resetSession(ExecConstants.HASHJOIN_ENABLE_RUNTIME_FILTER_KEY);
+    client.resetSession(ExecConstants.HASHJOIN_RUNTIME_FILTER_WAITING_ENABLE_KEY);
+  }
+
+  /**
+   * Test to make sure runtime filter is inserted in the plan. This is to ensure that with current cluster setup and
+   * system options distributed plan is generated rather than single fragment plan. Since in later case RuntimeFilter
+   * will not be inserted.
+   */
+  @Test
+  public void testRuntimeFilterPresentInPlan() throws Exception {
+    String sql = "SELECT l.n_name, r.r_name FROM dfs.`tpchmulti/nation` l, dfs.`tpchmulti/region/` r where " +
+      "l.n_regionkey = r.r_regionkey";
+    client.alterSession(ExecConstants.HASHJOIN_ENABLE_RUNTIME_FILTER_KEY, true);
+    String queryPlan = queryBuilder().sql(sql).explainText();
+    assertThat("Query plan doesn't contain RuntimeFilter. This may happen if plan is not distributed and has no " +
+        "exchange operator in it.", queryPlan, containsString("RuntimeFilter"));
+  }
+
+  /**
+   * Verifies that result of a query with join condition doesn't changes with and without Runtime Filter for correctness
+   */
+  @Test
+  public void testHashJoinCorrectnessWithRuntimeFilter() throws Exception {
+    String sql = "SELECT l.n_name, r.r_name FROM dfs.`tpchmulti/nation` l, dfs.`tpchmulti/region/` r where " +
+      "l.n_regionkey = r.r_regionkey";
+
+    String allTestQueryOptions = runtimeFilterOption + "; " + runtimeFilterWaitOption + "; " +
```

Comment:
I have provided a new constants with concatenation of all the options but will also like to keep the separate options so that it can be used for other tests separately if needed.